File: packages/core/android/src/main/java/expo/modules/gaodemap/overlays/LiveMarkerViewModule.kt
package expo.modules.gaodemap.overlays

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * LiveMarker 视图 Module
 */
class LiveMarkerViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("LiveMarkerView")

    View(LiveMarkerView::class) {
      Events("onPress")

      Prop<Map<String, Any>?>("coordinate") { view, position ->
        view.setPosition(position)
      }

      Prop<Map<String, Any>?>("anchor") { view, anchor ->
        view.setAnchor(anchor)
      }

      Prop<Map<String, Any>?>("offset") { view, offset ->
        view.setOffset(offset)
      }

      Prop<Int>("contentWidth") { view, width ->
        view.setContentWidth(width)
      }

      Prop<Int>("contentHeight") { view, height ->
        view.setContentHeight(height)
      }

      Prop<Boolean>("visible") { view, visible ->
        view.setVisible(visible)
      }

      Prop<Boolean>("tracksCamera") { view, enabled ->
        view.setTracksCamera(enabled)
      }

      Prop<Float>("zIndex") { view, zIndex ->
        view.setZIndexValue(zIndex)
      }
    }
  }
}

File: packages/core/ios/overlays/LiveMarkerViewModule.swift
import ExpoModulesCore

public class LiveMarkerViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("LiveMarkerView")

        View(LiveMarkerView.self) {
            Events("onPress")

            Prop("coordinate") { (view: LiveMarkerView, position: [String: Any]?) in
                view.setPosition(position)
            }

            Prop("anchor") { (view: LiveMarkerView, anchor: [String: Any]?) in
                view.setAnchor(anchor)
            }

            Prop("offset") { (view: LiveMarkerView, offset: [String: Any]?) in
                view.setOffset(offset)
            }

            Prop("contentWidth") { (view: LiveMarkerView, width: Int) in
                view.setContentWidth(width)
            }

            Prop("contentHeight") { (view: LiveMarkerView, height: Int) in
                view.setContentHeight(height)
            }

            Prop("visible") { (view: LiveMarkerView, visible: Bool) in
                view.setVisible(visible)
            }

            Prop("tracksCamera") { (view: LiveMarkerView, enabled: Bool) in
                view.setTracksCamera(enabled)
            }

            Prop("zIndex") { (view: LiveMarkerView, zIndex: Double) in
                view.setZIndex(zIndex)
            }
        }
    }
}
